code/go/0chain.net/smartcontract/zcnsc/authorizer.go

```diff
@@ -55,12 +64,13 @@ func (zcn *ZCNSmartContract) AddAuthorizer(tran *transaction.Transaction, inputD
 		return "", err
 	}
 
-	publicKey = params.PublicKey
+	authorizerPublicKey = params.PublicKey
+	authorizerURL = params.URL
 
 	// Check existing Authorizer
 
 	authorizer, err = GetAuthorizerNode(authorizerID, ctx)
-	if err == nil {
+	if err == nil && authorizer != nil {
```

Comment:
@peterlimg  I'm not following here: ```return error instead of continue```
In essence, I need to check whether AuthorizerNode exists.
Did you mean to update it here:

```go
	globalNode, err := GetGlobalNode(ctx)
	if err != nil {
		msg := fmt.Sprintf("failed to get global node, authorizer(authorizerID: %v), err: %v", authorizerID, err)
		err = common.NewError(code, msg)
		Logger.Error("get global node", zap.Error(err))
		return "", err
	}
```

Reply:
Fixed this